publish-to-npm.sh

```diff
@@ -1,5 +1,11 @@
 #!/bin/bash
 
+# Make sure local git is up-to-date.
+git fetch --all
+if [[ $(git rev-parse HEAD) != $(git rev-parse origin/master) ]]; then
```

Comment:
Are you willing to assume that your users always will be publishing the master branch? Would it be more appropriate to check $CURRENT_BRANCH against $ASSOCIATED_REMOTE_BRANCH?

Reply:
If not, it's probably worth adding additional checks to the effect of "Fail if the current branch is not `master`"